### ISSUE TITLE
label SparseTensor is not valid: indices[25] = [0,25] is out of bounds: need 0 <= index < [128,25]

### ISSUE THREAD

**Issue by ElagJun** [MaybeShewill-CV/CRNN_Tensorflow#63]
训练中文字符图片，报如下错误，请问这个是如何解决呢
``Caused by op 'CTCLoss', defined at:
  File "tools/train_shadownet.py", line 287, in <module>
    train_shadownet(args.dataset_dir, args.weights_path)
  File "tools/train_shadownet.py", line 80, in train_shadownet
    sequence_length=80*np.ones(config.cfg.TRAIN.BATCH_SIZE)))
  File "/home/admin/py3_tf120_gpu/lib/python3.4/site-packages/tensorflow/python/ops/ctc_ops.py", line 152, in ctc_loss
    ignore_longer_outputs_than_inputs=ignore_longer_outputs_than_inputs)
  File "/home/admin/py3_tf120_gpu/lib/python3.4/site-packages/tensorflow/python/ops/gen_ctc_ops.py", line 170, in _ctc_loss
    name=name)
  File "/home/admin/py3_tf120_gpu/lib/python3.4/site-packages/tensorflow/python/framework/op_def_library.py", line 767, in apply_op
    op_def=op_def)
  File "/home/admin/py3_tf120_gpu/lib/python3.4/site-packages/tensorflow/python/framework/ops.py", line 2506, in create_op
    original_op=self._default_original_op, op_def=op_def)
  File "/home/admin/py3_tf120_gpu/lib/python3.4/site-packages/tensorflow/python/framework/ops.py", line 1269, in __init__
    self._traceback = _extract_stack()

InvalidArgumentError (see above for traceback): label SparseTensor is not valid: indices[25] = [0,25] is out of bounds: need 0 <= index < [128,25]
	 [[Node: CTCLoss = CTCLoss[ctc_merge_repeated=true, ignore_longer_outputs_than_inputs=false, preprocess_collapse_repeated=false, _device="/job:localhost/replica:0/task:0/cpu:0"](shadow/cnn_subnetwork/LSTMLayers/transpose_time_major/_75, shuffle_batch/TakeManySparseFromTensorsMap, shuffle_batch/TakeManySparseFromTensorsMap:1, CTCLoss/sequence_length)]]
	 [[Node: edit_distance/_99 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/gpu:0", send_device="/job:localhost/replica:0/task:0/cpu:0", send_device_incarnation=1, tensor_name="edge_3391_edit_distance", tensor_type=DT_FLOAT, `_device="/job:localhost/replica:0/task:0/gpu:0"]()]]`

**Comment by Keysmis** [comment]
@ElagJun 请问你解决了这个问题了吗？我也遇到了同样的问题

**Comment by mdbenito** [comment]
This might be a mismatch between number of classes declared and used in the charsets. Newer versions of the code should fix it

**Comment by mdbenito** [comment]
I think this is indeed the problem I mention and has been addressed multiple times. @MaybeShewill-CV close?

**Comment by MaybeShewill-CV** [comment]
@mdbenito Yeah the problem has been mentioned several times. I will close it.
